testclock/clock.go

```diff
@@ -41,6 +41,16 @@ func (t *timer) Chan() <-chan time.Time {
 	return t.c
 }
 
+// alarm implements a mock clock.Alarm for testing purposes.
+type alarm struct {
+	*timer
+}
+
+// Reset is part of the clock.Timer interface.
+func (a *alarm) Reset(t time.Time) bool {
```

Comment:
Is `alarm` really just for "implementing a mock clock.Alarm for testing purposes"?
It seems that `resetTime` is all called here, so can't we just inline the code and avoid the extra func?

Reply:
Nope, the alarm implements a Reset that takes a time. The underlying timer implements everything else (but also implements a Reset that takes a duration).